src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -728,7 +756,85 @@ public URL getUrl() throws MalformedURLException {
         if (port == 0) {
             throw new IllegalStateException("This method must be called after calling #startJenkins.");
         }
-        return new URL("http", host, port, "/jenkins/");
+        return new URL(https ? "https" : "http", host, port, "/jenkins/");
+    }
+
+    /**
+     * Sets up HTTPS for the current instance, and disables plain HTTP.
+     * This generates a self-signed certificate for localhost. The corresponding root CA that needs to be trusted by HTTP client can be obtained using {@link #getRootCA()}.
+     *
+     * @return the current instance
+     * @see #createWebClient()
+     */
+    public RealJenkinsRule https() {
+        return https("localhost");
+    }
+
+
+    /**
+     * Sets up HTTPS for the current instance, and disables plain HTTP.
+     * This generates a self-signed certificate for the given host name. The corresponding root CA that needs to be trusted by HTTP client can be obtained using {@link #getRootCA()}.
+     * <p>
+     * You don't need to call {@link #withHost(String)} when calling this method.
```

Comment:
I think you should limit this to be either localhost or something matching `*.<list_of_domains>` and skip the SSL Cert creation at runtime, and ditch the bouncycastle dependency.

Reply:
Perhaps. Though we have had trouble in the past with test certs expiring and no one noticing until it became a nuisance.

Reply:
and if someone has a need for something that can not be solved like that - it would still seemingly need to be fixed so that tests would work relaible across different systems, so there could always be an option to have `withCertificate(Path cert)` and push the issue down to the plugin that is insisting on using some whacky name..